conda/cli/main_info.py

```diff
@@ -396,7 +391,91 @@ def builder():
         yield ("netrc file", info_dict["netrc_file"])
         yield ("offline mode", info_dict["offline"])
 
-    return "\n".join(("", *(f"{key:>23} : {value}" for key, value in builder()), ""))
+    return {key: value for key, value in builder()}
+
+
+def get_main_info_str(info_dict: dict[str, Any]) -> str:
+    """
+    Returns a printable string of the contents of ``info_dict``.
+
+    :param info_dict:  The output of ``get_info_dict()``.
+    :returns:  String to print.
+    """
+    display_info = get_main_info_display(info_dict)
+
+    return "\n".join(
+        ("", *(f"{key:>23} : {value}" for key, value in display_info.items()), "")
+    )
+
+
+InfoComponents = Literal["base", "channels", "envs", "system", "detail", "json_all"]
+
+
+def get_display_data(
```

Comment:
Let's refactor this in the way I suggested please, since it unnecessarily mixes the data to render with the style to render with in the return type.